### PR TITLE
Expand SSH Finding Model into multiple Categories

### PR DESCRIPTION
**Pull request by J12934**
## General Changes

- Change empty string fields like `hostname` and `server_banner` to null values
- The `location` of the findings will now be equal to the hostname (if existing), falling back to the `ip_address` if not set.
- All findings now have `hostname` and `ip_address` fields in their `attributes` map.
- Split out the finding categories. Currently all SSH Scanner findings are of the category `SSH Service`. This category will be kept for the general purpose informational finding. The other "policy violation" type findings will be moved into their own category. The changes to these two categories are grouped below

## Changes to `SSH Service` Category Findings

- Added extra attributes for supported `auth_methods`, `key_algorithms`, `encryption_algorithms`, `mac_algorithms` & `compression_algorithms` to easily see all relevant informations about the ssh server

## Changes to `Policy Violation` Type Findings

- These findings were previously also in the `SSH Service` Category but will now be moved into the `SSH Policy Violation` Category
- Which kind of policy was violated can be identified by the findings name. The following names are possible:
  - Insecure Methods / Algorithms:
    - `Discouraged SSH Authentication Method`
    - `Insecure SSH Key Algorithms`
    - `Insecure SSH Encryption Ciphers`
    - `Insecure SSH MAC Algorithms`
    - `Insecure SSH Compression Algorithms`
  - Missing Methods / Algorithms:
    - `Missing SSH Authentication Method`
    - `Missing SSH Key Algorithms`
    - `Missing SSH Encryption Ciphers`
    - `Missing SSH MAC Algorithms`
    - `Missing SSH Compression Algorithms`
  - `Outdated SSH Protocol Version`
- These findings have a `payload` field in their `attributes` map which indicates which kind of key algorithms, encryption ciphers or other are in violation of the policy.
- The description contains the original `ssh_scan` recommendation text.

### CODE REVIEW

**Comment by J12934** [comment]
I refuse to accept the last codacity issue, (that the `get_policy_violation_type` is too complex). I think their cyclomatic complexity calculation logic really doesn't like  case statements.

File: src/main.rb [hunk 1]
```diff
@@ -4,26 +4,24 @@
 Bundler.setup(:default)
 require 'ruby-scanner-scaffolding'
 require 'ruby-scanner-scaffolding/healthcheck'
-require_relative "./ssh_worker"
+require_relative './ssh_worker'
 
-set :port, 8080
+set :port, 8_080
```
Comment (rfelber) [inline src/main.rb:9]: Why that?

**Review by rfelber — APPROVED** [review]
Im not sure about the port change but beside that it seems to be good.

File: src/main.rb (hunk 1 above)
Comment (J12934) [reply src/main.rb:9]: Thats the recommended way to format larger numbers in ruby.
`_` is the thousands separator.

Comment (rfelber) [reply src/main.rb:9]: ok 👍

**Pull request merged**